### ISSUE TITLE
Embed Game links break on updating a HTML5 Game

### ISSUE THREAD

**Issue by 1bardesign** [itchio/itch.io#577]
The embed game links (like https://itch.io/embed-upload/348022) break upon uploading a new zip, or so it seems; is there a reason they don't get updated to point to a new one? I noticed this because it made updating my LD37 entry break the LD page for a little while before someone caught it.

I understand the rationale for a new embed link if it was a different zip, but when it's a clear replacement (and your site automatically replaces the old one and breaks the old link) it can lead to easy accidental link-rot!

**Comment by leafo** [comment]
I agree it's not clear at all. I'm thinking we need a separate kind of URL that points to the most recent upload for the game instead of the upload, but I'll keep the existing URLs around for backwards compatibility

**Comment by 1bardesign** [comment]
Both of those would be great.

As another, perhaps more general option, I think the "embed this specific file" thing would be great if it was just tied to the filename. I'm not sure of any legacy or technical reasons for using numeric codes, but `dev.itch.io/gamename/embed-upload/filename` would be welcome in my book. It's indeed a bit longer, but also clearer just on inspection which file it's pointing at, and therefore "makes sense" that it'd break if you change the filename or whatever.

`dev.itch.io/gamename/embed` or similar would indeed be useful as a "most recent, selected as the game zip" url - as I'd bet that's the most common use case by far :)

**Comment by tuzen** [comment]
+1  both options.

I believe that unintentional url change also defeats save games that use html local storage. 

Been fighting problems with link rot too. Sounds like the old workflow was designed to upload a unique zip, then update every embed, then delete old zip. I mistakenly expected the url would stay the same if I didn't change the zip filename.
